src/iceberg_functions/iceberg_multi_file_list.cpp

```diff
@@ -360,6 +361,14 @@ optional_ptr<const IcebergManifestEntry> IcebergMultiFileList::GetDataFile(idx_t
 				auto &data_files = manifest_file.data_files;
 				current_data_files.insert(current_data_files.end(), data_files.begin(), data_files.end());
 				transaction_data_idx++;
+			} else if (!transaction_delete_manifests.empty()) {
+				if (transaction_delete_idx >= transaction_delete_manifests.size()) {
+					//! Exhausted all the transaction-local data
+					return nullptr;
+				}
+				// add to the current data files? I don't think so.
+				auto &manifest_file = transaction_delete_manifests[transaction_delete_idx].get();
```

Comment:
This is a local variable, what is happening in this body that persists outside of this scope? Besides increasing `transaction_delete_idx`